plugins/kubernetes/app/models/kuber_deploy_service.rb

```diff
@@ -0,0 +1,58 @@
+require 'kubeclient'
+
+class KuberDeployService
+  attr_reader :kuber_release
+
+  delegate :build, :client, to: :kuber_release
+
+  def initialize(kuber_release)
+    @kuber_release = kuber_release
+  end
+
+  def deploy!
+    # TODO: handling different deploy strategies (rolling updates, etc.)
+    log 'starting deploy'
+
+    create_replication_controllers!
+    create_services!
+
+    log 'API requests complete'
+  end
+
+  def create_replication_controllers!
+    kuber_release.release_docs.each do |release_doc|
+      log 'creating ReplicationController', role: release_doc.kubernetes_role.name
+
+      rc = Kubeclient::ReplicationController.new(release_doc.rc_hash)
+      client.create_replication_controller(rc)
+    end
+
+    # Doing this for now to force the loading of these constants.
+    # I was getting a circular dependency error otherwise
+    Watchers::PodWatcher ; Watchers::ReplicationControllerWatcher ; Watchers::EventWatcher
```

Comment:
autoloading isn't thread safe, afaik. If you see this with eager loading on then it's a problem.